src/main/java/com/google/jenkins/flakyTestHandler/plugin/JUnitFlakyTestDataPublisher.java

```diff
@@ -43,15 +48,33 @@ public JUnitFlakyTestDataPublisher() {
   }
 
   @Override
-  public TestResultAction.Data getTestData(AbstractBuild<?, ?> abstractBuild, Launcher launcher,
-      BuildListener buildListener, TestResult testResult)
+  public TestResultAction.Data contributeTestData(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener, TestResult testResult)
       throws IOException, InterruptedException {
-    FlakyTestResult flakyTestResult = launcher.getChannel().call(new FlakyTestResultCollector(testResult));
+    VirtualChannel channel = launcher.getChannel();
+    if(channel == null) {
+          throw new InterruptedException("Could not get channel to run a program remotely.");
+    }
+    FlakyTestResult flakyTestResult = channel.call(new FlakyTestResultCollector(testResult));
     // TODO consider the possibility that there is >1 such action
-    flakyTestResult.freeze(abstractBuild.getAction(AbstractTestResultAction.class), abstractBuild);
+    flakyTestResult.freeze(run.getAction(AbstractTestResultAction.class), run);
     return new JUnitFlakyTestData(flakyTestResult);
   }
 
+  @Override
+  public TestResultAction.Data getTestData(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener, TestResult testResult) {
```

Comment:
getTestData method is deprecated and the base class already provide call contributeTestData method